### DISCUSSION TITLE
How to assign bindings in javascript

### DISCUSSION THREAD

**Discussion by basestuff** [Budibase/budibase#10718] [Help]
I have a small JavaScript Snippet that creates a downloadable _**.vcf**_ file using some data in the app. The snippet runs perfectly with and without HTML, but I cannot find out why it has an issue running in budibase.

**### Edit:** The formatting of the code with code tags dont work so I will add an image:

**This is the basic JS code that works fine:**
![Screenshot 2023-05-26 230234](https://github.com/Budibase/budibase/assets/134717162/e324c425-2349-4e21-87da-77388476a444)

**This is my version for BudiBase using bindings:**
![Screenshot 2023-05-26 230234](https://github.com/Budibase/budibase/assets/134717162/90eca30f-d4cd-460e-bfe3-af6878db06ce)

Link to similar snippet that works: https://dev.to/walternascimentobarroso/make-vcard-with-js-2afl

The errors don't give me much to work on. Any guidance would be appreciated to understand this more. I have read through the docs but I am missing something somewhere. Thanks

**Comment by melohagan** [comment]
Hey @basestuff 

A couple of thoughts:

1. Budibase JavaScript binding **must return a value**. So the last line should be `return makeVCard();` for example.
2. I don't believe you can access the global variables within a Budibase JavaScript binding. 

You may need to take a look at the [Embed](https://docs.budibase.com/docs/embed) component for this one, or perhaps a [custom component](https://docs.budibase.com/docs/custom-component)
